### PR TITLE
feat(notification): Notifications pass through className, style, and id

### PR DESCRIPTION
**Pull request by matt-royal**
Notification follows dropdown semantics for passing through these
attributes.

[Finishes #98913374]

### CODE REVIEW

**Comment by gpleiss** [comment]
`pui-react-helpers` needs to be added to the package.json for notifications

File: spec/pivotal-ui-react/notification/notification_spec.js [hunk 1]
```diff
@@ -14,18 +29,32 @@ describe('Notification', function() {
...
+
+    it('passes through id to the button', function() {
+      expect('#root .btn#test-id').toExist();
+    });
+
     it('renders a notification count badge', function() {
       expect('.dropdown-notifications-title .dropdown-notifications-badge').toContainText('4');
     });
 
+    itPropagatesAttributes('#root li:first', itemProps);
```
Comment (gpleiss) [inline spec/pivotal-ui-react/notification/notification_spec.js]: being picky here: can we drag this out into a `describe` block for `NotificationItem`? It's confusing seeing it here.

File: spec/pivotal-ui-react/notification/notification_spec.js [hunk 2]
```diff
@@ -14,18 +29,32 @@ describe('Notification', function() {
...
+    });
+
+    it('passes through style to the button', function() {
+      expect('#root .btn').toHaveCss(props.style);
+    });
+
+    it('passes through id to the button', function() {
+      expect('#root .btn#test-id').toExist();
+    });
+
```
Comment (gpleiss) [inline spec/pivotal-ui-react/notification/notification_spec.js]: Given that id and style are not propagated in the standard way, we should probably document this.

**Comment by gpleiss** [comment]
Given that our notification component is a thin wrapper around Dropdowns, and we have a `buttonClassName` property on our dropdowns, do we want this property to be available for Notifications as well? I could see a user wanting to change the color of the notification bell, and they'd need this property to do so.

cc/ @stubbornella

**Comment by gpleiss** [comment]
In tracker, it says to "confirm this works for Notifications and AlertNotifications". I only see tests for Notifications.

**Comment by gpleiss** [comment]
We've addressed these comments.

@stubbornella ping

**Comment by matt-royal** [comment]
I just pushed an updated version of this branch that can be merged.

**Pull request merged**